SIL.Archiving.Tests/IMDIArchivingDlgViewModelTests.cs

```diff
@@ -263,4 +265,123 @@ public void SetAbstract_MultipleLanguages_AddsDescriptionToCorpusImdiFile()
+			model.OverrideGetFileGroupDisplayMessage = s => throw new AssertionException(
+				$"{nameof(ArchivingDlgViewModel.OverrideGetFileGroupDisplayMessage)} should not have been invoked");
+
+			model.InitializationFailed += (sender, e) => Assert.Fail("Initialization failed");
+
+			try
+			{
+				await model.Initialize(progress, new CancellationToken()).ConfigureAwait(false);
+			}
+			catch (Exception ex)
+			{
+				Assert.Fail($"Initialization threw an exception: {ex}");
+			}
+
+			Assert.True(filesToArchiveCalled);
+			Assert.That(customSummaryShown, Is.EqualTo(1));
+			Assert.That(progress.Step, Is.EqualTo(1));
+		}
+	}
+
+	[TestFixture]
+	[Category("Archiving")]
+	public class IMDIArchivingDlgViewModelWithFineGrainedOverridesForDisplayInitialSummarySetTests
+	{
+		/// ------------------------------------------------------------------------------------
+		[Test]
+		public async Task DisplayInitialSummary_OverridenPropertiesForDisplayInitialSummaryAreSet_MessagesReflectOverrides()
+		{
+			Console.WriteLine($"IMDI Tests TEMP: {nameof(DisplayInitialSummary_OverridenPropertiesForDisplayInitialSummaryAreSet_MessagesReflectOverrides)}");
+
+			ErrorReport.IsOkToInteractWithUser = false;
+
+			void SetFilesToArchive(ArchivingDlgViewModel model, CancellationToken cancellationToken)
+			{
+				model.AddFileGroup(String.Empty, new[] { "green.frog" }, "These messages should not be displayed");
+				model.AddFileGroup("Toads", new[] { "red.toad", "blue.toad" }, "because in this test we do not create a package.");
+			}
+
+			var model = new IMDIArchivingDlgViewModel("Test App", "Test Title", "tst", true,
+				SetFilesToArchive, "unused");
+
+			var messagesDisplayed = new List<Tuple<string, ArchivingDlgViewModel.MessageType>>();
+
+			void ReportMessage(string msg, ArchivingDlgViewModel.MessageType type)
+			{
+				messagesDisplayed.Add(new Tuple<string, ArchivingDlgViewModel.MessageType>(msg, type));
+			}
+
+			model.OnReportMessage += ReportMessage;
+
+			IEnumerable<Tuple<string, ArchivingDlgViewModel.MessageType>> GetMessages(IDictionary<string, Tuple<IEnumerable<string>, string>> arg)
+			{
+				yield return new Tuple<string, ArchivingDlgViewModel.MessageType>(
+					"First pre-archiving message", Warning);
+				yield return new Tuple<string, ArchivingDlgViewModel.MessageType>(
+					"Second pre-archiving message", Indented);
+			}
+
+			model.GetOverriddenPreArchivingMessages = GetMessages;
+			model.InitialFileGroupDisplayMessageType = Success;
+			model.OverrideGetFileGroupDisplayMessage = s => (s == String.Empty) ? "Frogs" : $"Label: {s}";
+			model.InitializationFailed += (sender, e) => Assert.Fail("Initialization failed");
+
+			var progress = new TestProgress("IMDI");
+			try
+			{
+				await model.Initialize(progress, new CancellationToken()).ConfigureAwait(false);
+			}
+			catch (Exception ex)
+			{
+				Assert.Fail($"Initialization threw an exception: {ex}");
+			}
+
+			Assert.That(messagesDisplayed.Count, Is.EqualTo(7));
```

Comment:
Might I suggest something like this instead
```c#
			Assert.That(messagesDisplayed, Is.EqualTo(new[]
			{
				("First pre-archiving message", Warning).ToTuple(),
				("Second pre-archiving message", Indented).ToTuple(),
				("Frogs", Success).ToTuple(),
				("green.frog", Bullet).ToTuple(),
				("Label: Toads", Success).ToTuple(),
				("red.toad", Bullet).ToTuple(),
				("blue.toad", Bullet).ToTuple()
			}));
```

Reply:
Done